accepted/0000-replace-buildpack-yml.md

```diff
@@ -0,0 +1,52 @@
+# Replace buildpack.yml with Build Plan (TOML)
+
+## Summary
+
+`buildpack.yml` is a Paketo specific construct that requires increased overhead for buildpack authors. Everything that is done in `buildpack.yml` can be done with the [`build-plan` buildpack](https://github.com/ForestEckhardt/build-plan), which would offload some logic onto the lifecycle.
+
+## Motivation
+
+With this change we can completely eliminate `buildpack.yml` which is a file that adds undue complexity to the buildpacks because in order for `buildpack.yml` influence what appears in the application image it must be parsed during detection in every buildpack. `buildpack.yml` also has a non-standard format meaning that every single buildpack needs to have slightly unique parsing logic in order to grab a buildpack specific key name and then any other additional metadata.
+
+The big advantage of moving to using `build-plan` is that every buildpack that currently has a `buildpack.yml` parser could remove that code entirely and switch to using Buildpack Plan metadata during the `build` phase and there would be no loss of functionality. The job of parsing build configuration will be condensed into the `build-plan` and then all subsequent parsing will be handled by the buildpack library or the lifecycle.
+
+The format used in `build-plan` is also more powerful than `buildpack.yml` because it allows for fine grain controls as to when that dependency is available by setting `build` and `launch` flags. Any other data that is not the dependency name or version can be put in the metadata section of requires and handled during the `build` phase.
+
+By using the `build-plan` buildpack we can completely replace the `buildpack.yml` while losing no functionality, increase user configurability with very low overhead cost for buildpack authors, and vastly increased interoperability.
+
+The ideal end state for this track or work would be to bake this functionality directly into the lifecycle because of how closely the `build-plan` buildpack already interacts with the lifecycle. The `build-plan` buildpack is a middle ground to test out whether or not this model is a good thing for users and buildpack authors. Once this model has proven (or disproven) itself, learnings from this stop gap should be used to inform further actions (whether that be pursuing integration in the lifecycle or abandoning this model for something else).
```

Comment:
I'm not totally sold on this being a goal for the CNB spec.  At this time, we don't have a single situation where the `lifecycle` knows about a file in the source that it is transforming, and this would cause that to happen.  I think this may end up being part of platform implementations, but this feels like we're getting ahead of ourselves.

/cc @ekcasey

Reply:
I agree that this is out of the purview of the this particular RFC. I think I should move it to a different section that is `Potential Long-Term Goals`. This was added because of some prior feedback about how this RFC could be extended but I think that it is in an inappropriate section to indicate as such.

Reply:
I have moved this point into a section at the bottom titled `Potential Long-Term Goals`.